tests/integration/test_k8s.py

```diff
@@ -83,6 +86,48 @@ async def test_nodes_ready(kubernetes_cluster: model.Model):
     await ready_nodes(k8s.units[0], expected_nodes)
 
 
+@retry(reraise=True, stop=stop_after_attempt(12), wait=wait_fixed(60))
+async def test_grafana(
+    traefik_address: str,
+    grafana_password: str,
+    expected_dashboard_titles: set,
+    cos_model: model.Model,
+):
+    """Test integration with Grafana."""
+    grafana = Grafana(model_name=cos_model.name, host=traefik_address, password=grafana_password)
+    while not await grafana.is_ready():
+        log.info("Waiting for Grafana to be ready.")
+        await asyncio.sleep(5)
```

Comment:
So you could use `wait_for` here i think -- but it wouldn't have the sleep and log.  maybe this is okay -- but there's no timeout

```suggestion
    await asyncio.wait_for(grafana.is_ready(), timeout=5*60)
```